File: src/main/java/net/kuryshev/webnotes/NoteController.java
package net.kuryshev.webnotes;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api")
public class NoteController {

    @Autowired
    private NoteRepository noteRepository;

    @GetMapping("/note/{id}")
    public Note getNote(@PathVariable Long id) {
        return noteRepository.getOne(id);
    }

    @GetMapping("/notes")
    public List<Note> getNotes() {
        return noteRepository.findAll();
    }

    @PostMapping("/note")
    public Long saveNote(@RequestBody Note note) {
        return noteRepository.save(note).getId();
    }

    @DeleteMapping("/note/{id}")
    public void deleteNote(@PathVariable Long id) {
        noteRepository.deleteById(id);
    }
}

File: api/src/main/java/net/kuryshev/webnotes/note/NoteController.kt
package net.kuryshev.webnotes.note

import net.kuryshev.webnotes.common.SuccessResponse
import org.springframework.web.bind.annotation.*
import java.security.Principal

@RestController
@RequestMapping("/api")
class NoteController(val noteService: NoteService) {
    @GetMapping("/note/{id}")
    fun getTask(@PathVariable id: Long, principal: Principal) = noteService.getNoteForUser(id, principal.name)

    @GetMapping("/notes")
    fun getTasks(principal: Principal) = noteService.getNotesForUser(principal.name)

    @PostMapping("/note")
    fun saveTask(@RequestBody note: Note, principal: Principal) = noteService.saveNoteForUser(note, principal.name)

    @DeleteMapping("/note/{id}")
    fun deleteTask(@PathVariable id: Long, principal: Principal) = SuccessResponse(noteService.deleteNoteForUser(id, principal.name))
}